assertion/function/assertiontree/backprop_util.go

```diff
@@ -172,7 +160,16 @@ func computeAndConsumeResults(rootNode *RootAssertionNode, node *ast.ReturnStmt)
 						}
 						fullTrigger := annotation.FullTrigger{
 							Producer: producer,
-							Consumer: consumer,
+							Consumer: &annotation.ConsumeTrigger{
+								Annotation: &annotation.UseAsReturn{
+									TriggerIfNonNil: &annotation.TriggerIfNonNil{
+										Ann: retKey},
+									IsNamedReturn: true,
+									RetStmt:       node,
+								},
+								Expr:   retVariable,
+								Guards: util.NoGuards(),
+							},
```

Comment:
this is because blank identifier will be nilable by default and hence we don't need to track its deep nilabilities, right?

The code is a bit messy here tbh, but let's refactor that later

Reply:
That's correct.

I know, like many other places the code here could do with refactoring and reorganizing as well :D